pallet/src/tests/signer.rs

```diff
@@ -346,3 +292,185 @@ fn script_with_vector_containing_signer_fails() {
+    let (bob_addr_32, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (dave_addr_32, dave_addr_mv) = addrs_from_ss58(DAVE_ADDR).unwrap();
+    let (eve_addr_32, eve_addr_mv) = addrs_from_ss58(EVE_ADDR).unwrap();
+
+    ExtBuilder::default()
+        .with_balances(vec![
+            (bob_addr_32.clone(), BALANCE),
+            (alice_addr_32.clone(), BALANCE),
+            (dave_addr_32.clone(), BALANCE),
+            (eve_addr_32.clone(), BALANCE),
+        ])
+        .build()
+        .execute_with(|| {
+            // Roll to first block in case of block based event checkings and processes.
+            roll_to(1);
+
+            // Initialisation & Setup by developer Bob.
+            let module = assets::read_module_from_project("multiple-signers", "Dorm");
+            assert_ok!(MoveModule::publish_module(
+                RuntimeOrigin::signed(bob_addr_32.clone()),
+                module,
+                MAX_GAS_AMOUNT
+            ));
+            let script = assets::read_script_from_project("multiple-signers", "init_module");
+            let transaction_bc = script_transaction!(script, no_type_args!(), &bob_addr_mv);
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(bob_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+
+            // Now only 2 of 3 planned signers will sign the script execution.
+            let script = assets::read_script_from_project("multiple-signers", "rent_apartment");
+            let transaction_bc = script_transaction!(
+                script,
+                no_type_args!(),
+                &alice_addr_mv,
+                &dave_addr_mv,
+                &eve_addr_mv
+            );
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(alice_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+            assert_ok!(MoveModule::execute(
+                RuntimeOrigin::signed(dave_addr_32.clone()),
+                transaction_bc.clone(),
+                MAX_GAS_AMOUNT,
+                BALANCE,
+            ));
+            // Sloppy or distrustful Eve is missing...
+
+            // Verify expected Multisig data in storage.
+            let call_hash = MoveModule::transaction_bc_call_hash(&transaction_bc[..]);
+            let _request = MultisigStorage::<Test>::try_get(call_hash).unwrap();
+
+            // Let's roll forward to block number 4 and check our request still exists.
+            roll_to(5);
+            assert!(MultisigStorage::<Test>::try_get(call_hash).is_ok());
+
+            // One more block forward and it shall be removed!
+            roll_to(6);
+            assert!(MultisigStorage::<Test>::try_get(call_hash).is_err());
```

Comment:
We could (in future) add the case where Eve tries to sign it after and then nothing happens because we are then missing the first two signers (expired)

Reply:
In that case, she would create a new multi-signer request - she will get the even `SignedMultisigScript`. Not sure, if we can detect, that she tried to sign an outdated request :thinking:

Reply:
Added this test detail too :+1: